nomad/update.go

```diff
@@ -0,0 +1,128 @@
+package nomad
+
+import (
+	"context"
+	"io"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"github.com/hashicorp/nomad/api"
+	"github.com/pkg/errors"
+	"github.com/rs/zerolog/log"
+)
+
+func Update(ctx context.Context, nomadCli *api.Client, job *api.Job, task *api.Task, imageName string, scheduleId string) error {
+
+	log.Info().
+		Str("image", imageName).
+		Str("task", task.Name).
+		Str("schedule-id", scheduleId).
+		Interface("task config", task.Config).
+		Msg("Updating Portainer agent")
+
+	job.Update = api.DefaultUpdateStrategy()
+	task.Config["image"] = imageName
+
+	response, _, err := nomadCli.Jobs().EnforceRegister(job, *job.JobModifyIndex, nil)
+	if err != nil {
+		return errors.WithMessage(err, "failed to register job")
+	}
+
+	log.Debug().
+		Str("job", *job.Name).
+		Str("warnings", response.Warnings).
+		Msg("Job registered")
+
+	allocations, _, err := nomadCli.Jobs().Allocations(*job.ID, false, &api.QueryOptions{WaitIndex: response.JobModifyIndex})
+	if err != nil {
+		return errors.WithMessage(err, "failed to get allocations for job")
+	}
+
+	for _, allocation := range allocations {
+		if allocation.ClientStatus != api.AllocClientStatusPending {
+			continue
+		}
+
+		log.Info().
+			Str("allocation", allocation.ID).
+			Msg("waiting for allocation to start")
+
+		for {
+			time.Sleep(1 * time.Second)
+
+			log.Debug().
+				Str("allocation", allocation.ID).
+				Msg("polling allocation")
+
+			allocation, _, err := nomadCli.Allocations().Info(allocation.ID, nil)
+			if err != nil {
+				return errors.WithMessage(err, "failed to get allocation info")
+			}
+
+			if allocation.ClientStatus == api.AllocClientStatusPending {
+				continue
+			}
+
+			if allocation.ClientStatus == api.AllocClientStatusRunning {
+				log.Debug().
```

Comment:
this should probably be converted to `Info` to let the user know that update suceed

Reply:
fixed